London | Funmi Moriyele | Wireframe Task | Week 2

fmoriyele opened the pull request:
**Self checklist**
✓ I have committed my files one by one, on purpose, and for a reason
✓ I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK
✓ I have tested my changes
✓ My changes follow the [style guide](https://syllabus.codeyourfuture.io/guides/code-style-guide/)
✓ My changes meet the [requirements](https://app.slack.com/client/T07MCNDRDHV/README.md) of this task

**Changelist**
✓ I have created a webpage about Git that contains three articles. I used HTML5, CSS3, and Web Design to develop this webpage.

**Questions**

I struggled to use the :first-class pseudo class. I applied this to Article 1, creating a `<div class="article-section">`  to make sure Article 1 is the first child of the parent container. However, it seems to make no changes to the code when I use css to style it.

I would appreciate any feedback on how to correct this.

Thank you for reviewing!

fmoriyele committed:
Skeleton page code, no styling added

fmoriyele committed:
Adding text information and images to the page title, articles and footer.

fmoriyele committed:
Flex-box container added for last two articles.
Classes added for styling

fmoriyele committed:
Styling added to page

fmoriyele committed:
Fixing footer to bottom of viewport

SallyMcGrath reviewed (changes requested):
Hi @fmoriyele . Thanks for this! Your work is deployed here https://deploy-preview-206--cyf-ufd.netlify.app/wireframe/

1. When I ran Lighthouse it scored 83. Run Lighthouse yourself and make the suggested changes until your score is 100

2. Regarding your question. You don't need this div and it's complicating your code (nearly always in HTML and CSS the answer is to remove code) . Put all three articles as child of main and then try assigning a style to article:first-child . 

https://developer.mozilla.org/en-US/docs/Web/CSS/:first-child

fmoriyele committed:
Using first-child pseudoclass

fmoriyele committed:
Centering first article

fmoriyele committed:
Adding image descriptions to the three article images

fmoriyele committed:
Made buttons larger for accessibility (Lighthouse)

fmoriyele commented:
Hi, I have followed the Lighthouse suggested and added image descriptions for my images.
I also made the buttons larger and added more padding. However, no matter how large I make the buttons, Lighthouse does not seem to recognise it.

I also used the article:first-class pseudoclass which worked but seemed to include the flexbox first child as well (article 2). I have made a commit named "using first-class pseudoclass" where I highlighted in red the item the code selects as the first child.

I made my following changes by targeting the first article using an ID classifier.

SallyMcGrath committed:
Merge branch 'main' into feature/wireframe

SallyMcGrath commented:
> Hi, I have followed the Lighthouse suggested and added image descriptions for my images. I also made the buttons larger and added more padding. However, no matter how large I make the buttons, Lighthouse does not seem to recognise it.
> 

It's because you are combining buttons and anchors. You can only have one - they don't combine. A button is for triggering an action (like maybe "Comment") and an anchor (link)  is for navigation. You cannot nest them inside each other. Links and buttons are often styled to look the same, but underneath they have different functions and features. 

> I also used the article:first-class pseudoclass which worked but seemed to include the flexbox first child as well (article 2). I have made a commit named "using first-class pseudoclass" where I highlighted in red the item the code selects as the first child.

Yes, you're nearly there! But what I'm saying is you don't need this extra div at all. It's a really common thing for junior devs to do -- add loads and loads of pointless divs everywhere, but it actually makes your code just more difficult to manage.  Set up a `<main>` element and put all three articles as direct children of main. Then the first article will be the first child. 

When you add in that extra div, you make the first article the first child of its parent, and the second article the first child of that extra div you added.  CSS Diner might help you with this https://flukeout.github.io/
> 
> I made my following changes by targeting the first article using an ID classifier.

You will almost never use an id for CSS rules. Look into specificity to see why this is, and maybe Khan Academy can be helpful here https://www.khanacademy.org/computing/computer-programming/html-css/
